From 26d210e313b1d2dc034daa4833ccd9823e3240bd Mon Sep 17 00:00:00 2001
From: Jianxu Zheng <jian.xu.zheng@intel.com>
Date: Thu, 26 Mar 2015 10:22:35 +0800
Subject: [PATCH 322/323] ov8858: fix absolute exposure v4l2_ctrl

for absolute exposure, ctrl type should be integer not menu.
Also several setting change related to integer type v4l2_ctrl.

Change-Id: I536682af81a3144aa8538ea74730f77ae48e0476
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-27398
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/47897
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index d154e03c..6196108 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -2078,9 +2078,12 @@ static const struct v4l2_ctrl_config ctrls[] = {
 		.ops = &ctrl_ops,
 		.id = V4L2_CID_EXPOSURE_ABSOLUTE,
 		.name = "Absolute exposure",
-		.type = V4L2_CTRL_TYPE_MENU,
+		.type = V4L2_CTRL_TYPE_INTEGER,
 		.max = 0xffff,
-		.qmenu = ctrl_run_mode_menu,
+		.min = 0x0,
+		.step = 1,
+		.def = 0x00,
+		.flags = 0,
 	}, {
 		.ops = &ctrl_ops,
 		.id = V4L2_CID_TEST_PATTERN,
-- 
1.9.1

